src/sonic-yang-models/yang-models/sonic-device_metadata.yang

```diff
@@ -0,0 +1,95 @@
+module sonic-device_metadata {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-device_metadata";
+    prefix device_metadata;
+
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-types {
+        prefix stypes;
+        revision-date 2019-07-01;
+    }
+
+    description "DEVICE_METADATA YANG Module for SONiC OS";
+
+    revision 2020-04-10 {
+        description "First Revision";
+    }
+
+    container sonic-device_metadata {
+
+        container DEVICE_METADATA {
+
+            description "DEVICE_METADATA part of config_db.json";
+
+            container localhost{
+
+                leaf hwsku {
+                    type stypes:hwsku;
+                }
+
+                leaf default_bgp_status {
+                    type enumeration {
+                        enum up;
+                        enum down;
+                    }
+                }
+
+                leaf docker_routing_config_mode {
+                    type string {
+                        pattern "unified|split|separated";
+                    }
+                }
+
+                leaf hostname {
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf platform {
+                    type string {
+                        length 1..255;
+                    }
+                }
+
+                leaf mac {
+                    type yang:mac-address;
+                }
+
+                leaf default_pfcwd_status {
+                    type enumeration {
+                        enum disable;
+                        enum enable;
+                    }
+                }
+
+                leaf bgp_asn {
+                    type inet:as-number;
+                }
+
+                leaf deployment_id {
+                    type uint32;
+                }
+
+                leaf type {
+                    type string {
+                        length 1..255;
```

Comment:
should be enum.

Reply:
@lguohan  Any links on what are the possible values for type?, thx.

Reply:
ToRRouter, LeafRouter, SpineChassisFrontendRouter, ChassisBackendRouter, ASIC

Reply:
I think Pattern is better when enum fields are more. So added that.